src/token.rs

```diff
@@ -13,6 +13,52 @@ pub struct Span {
     pub end: u32,
 }
 
+pub trait Spanner {
+    fn get_span(&self) -> Span;
```

Comment:
`get_foo()` is not very idomatic; just `foo()` is preferred. Could you use `fn span()`?